File: ALittle/Module/ALittleIDE/Script/UIEdit/IDEEditCore/IDESetting/TextS.lua

module("ALittleIDE", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs
local ___coroutine = coroutine


assert(DisplayObjectS, " extends class:DisplayObjectS is nil")
TextS = ALittle.Class(DisplayObjectS, "ALittleIDE.TextS")

function TextS:Ctor(user_info, tab_child, tree_logic)
	___rawset(self, "_layer_name", "ide_setting_text")
end

function TextS:LoadNatureBase()
	DisplayObjectS.LoadNatureBase(self)
	self:LoadBoolData("bold", false, g_IDEEnum.yn_type)
	self:LoadBoolData("underline", false, g_IDEEnum.yn_type)
	self:LoadBoolData("outline", false, g_IDEEnum.yn_type)
	self:LoadBoolData("deleteline", false, g_IDEEnum.yn_type)
	self:LoadBoolData("italic", false, g_IDEEnum.yn_type)
	self:LoadValueData("text")
	self:LoadDefaultNilString("font_path")
	self:LoadDefaultNilString("font_size")
	self:LoadEnumData("flip", g_IDEEnum.flip_type)
end

function TextS:HandleBoldSELECT_CHANGE(event)
	self:BoolSelectChange("bold", false, g_IDEEnum.yn_type)
end

function TextS:HandleUnderLineSELECT_CHANGE(event)
	self:BoolSelectChange("underline", false, g_IDEEnum.yn_type)
end

function TextS:HandleOutLineSELECT_CHANGE(event)
	self:BoolSelectChange("outline", false, g_IDEEnum.yn_type)
end

function TextS:HandleDeleteLineSELECT_CHANGE(event)
	self:BoolSelectChange("deleteline", false, g_IDEEnum.yn_type)
end

function TextS:HandleItalicSELECT_CHANGE(event)
	self:BoolSelectChange("italic", false, g_IDEEnum.yn_type)
end

function TextS:HandleTextFOCUSOUT(event)
	if self._default.__class == "Text" then
		self:DefaultNoStringInputChange("text", true)
	else
		self:DefaultNoStringInputChange("text", false)
	end
end

function TextS:HandleFontPathFOCUSOUT(event)
	if self._default.__class == "Text" then
		self:SpecialFontPathInputChange("font_path", true)
	else
		self:SpecialFontPathInputChange("font_path", false)
	end
end

function TextS:HandleFontSizeFOCUSOUT(event)
	if self._default.__class == "Text" then
		self:ValueNumZInputChange("font_size", true)
	else
		self:ValueNumZInputChange("font_size", false)
	end
end

function TextS:HandleFlipSELECT_CHANGE(event)
	local list = g_IDEEnum.flip_rtype
	self:TypeSelectChange("flip", list, false)
end

File: ALittle/Module/ALittleIDE/JSScript/IDE/UIEdit/Setting/TextS.js
{
if (typeof ALittleIDE === "undefined") window.ALittleIDE = {};


if (ALittleIDE.DisplayObjectS === undefined) throw new Error(" extends class:ALittleIDE.DisplayObjectS is undefined");
ALittleIDE.TextS = JavaScript.Class(ALittleIDE.DisplayObjectS, {
	Ctor : function(user_info, tab_child, tree_logic) {
		this._layer_name = "ide_setting_text";
	},
	LoadNatureBase : function() {
		ALittleIDE.DisplayObjectS.LoadNatureBase.call(this);
		this.LoadBoolData("bold", false, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadBoolData("underline", false, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadBoolData("deleteline", false, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadBoolData("italic", false, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadValueData("text");
		this.LoadDefaultNilString("font_path");
		this.LoadDefaultNilString("font_size");
		this.LoadEnumData("flip", ALittleIDE.g_IDEEnum.flip_type);
	},
	HandleBoldSELECT_CHANGE : function(event) {
		this.BoolSelectChange("bold", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleUnderLineSELECT_CHANGE : function(event) {
		this.BoolSelectChange("underline", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleDeleteLineSELECT_CHANGE : function(event) {
		this.BoolSelectChange("deleteline", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleItalicSELECT_CHANGE : function(event) {
		this.BoolSelectChange("italic", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleTextFOCUSOUT : function(event) {
		if (this._default.__class === "Text") {
			this.DefaultNoStringInputChange("text", true);
		} else {
			this.DefaultNoStringInputChange("text", false);
		}
		this._tree_logic.UpdateDesc();
	},
	HandleFontPathFOCUSOUT : function(event) {
		if (this._default.__class === "Text") {
			this.SpecialFontPathInputChange("font_path", true);
		} else {
			this.SpecialFontPathInputChange("font_path", false);
		}
	},
	HandleFontSizeFOCUSOUT : function(event) {
		if (this._default.__class === "Text") {
			this.ValueNumZInputChange("font_size", true);
		} else {
			this.ValueNumZInputChange("font_size", false);
		}
	},
	HandleFlipSELECT_CHANGE : function(event) {
		let list = ALittleIDE.g_IDEEnum.flip_rtype;
		this.TypeSelectChange("flip", list, false);
	},
}, "ALittleIDE.TextS");

}